packages/runtime/container-runtime/src/summarizerTypes.ts

```diff
@@ -34,6 +33,12 @@ export interface IProvideSummarizer {
     readonly ISummarizer: ISummarizer;
 }
 
+/* Similar to AbortSignal, but using promise instead of events */
```

Comment:
why are promises better than events? i think i'd prefer to just have an interface that maps to abort signal:
https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal

it's pretty trivial to add a wrapper over the event if you want a promise.

Reply:
Promises compose easier. I.e. 
Promise.race([ something I wait for, cancellable.waitCacenlled]);

That same thing becomes a lot of lines of code if I were using signals.
At the end of the day, I'd write converter from abort signal to promise and would use it :)